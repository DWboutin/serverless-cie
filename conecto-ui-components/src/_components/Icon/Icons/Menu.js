import React from 'react'

const Menu = () => (
  <div className="to-icon-menu">
    <span />
    <span />
  </div>
)

export default Menu
