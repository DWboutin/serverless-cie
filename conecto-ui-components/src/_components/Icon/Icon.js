import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import To from '../Logo/Logos/To'
import Menu from './Icons/Menu'
import Alert from './Icons/Alert'
import ArrowDown from './Icons/ArrowDown'
import Bookmark from './Icons/Bookmark'
import Check from './Icons/Check'
import Close from './Icons/Close'
import Cog from './Icons/Cog'
import Concurrents from './Icons/Concurrents'
import Email from './Icons/Email'
import Facebook from './Icons/Facebook'
import Flag from './Icons/Flag'
import ColoredFlag from './Icons/ColoredFlag'
import Marker from './Icons/Marker'
import Notification from './Icons/Notification'
import Phone from './Icons/Phone'
import Search from './Icons/Search'
import SMS from './Icons/SMS'
import ThumbUp from './Icons/ThumbUp'
import User from './Icons/User'

import CCAmex from './Icons/CCAmex'
import CCDefault from './Icons/CCDefault'
import CCDinersClub from './Icons/CCDinersClub'
import CCDiscover from './Icons/CCDiscover'
import CCJCB from './Icons/CCJCB'
import CCMastercard from './Icons/CCMastercard'
import CCVisa from './Icons/CCVisa'

class Icon extends React.Component {
  constructor(props) {
    super(props)

    this.classesPrefix = 'to-icon'
  }

  iconDefinition(type) {
    switch (type) {
      case 'alert':
        return Alert
      case 'arrow-down':
        return ArrowDown
      case 'bookmark':
        return Bookmark
      case 'check':
        return Check
      case 'close':
        return Close
      case 'cog':
        return Cog
      case 'concurrents':
        return Concurrents
      case 'email':
        return Email
      case 'facebook':
        return Facebook
      case 'flag':
        return Flag
      case 'colored-flag':
        return ColoredFlag
      case 'marker':
        return Marker
      case 'menu':
        return Menu
      case 'notification':
        return Notification
      case 'phone':
        return Phone
      case 'search':
        return Search
      case 'sms':
        return SMS
      case 'thumb-up':
        return ThumbUp
      case 'user':
        return User
      case 'cc-amex':
        return CCAmex
      case 'cc-default':
        return CCDefault
      case 'cc-dinersclub':
        return CCDinersClub
      case 'cc-discover':
        return CCDiscover
      case 'cc-jcb':
        return CCJCB
      case 'cc-mastercard':
        return CCMastercard
      case 'cc-visa':
        return CCVisa
      case 'conecto':
      default:
        return To
    }
  }

  render() {
    const { className, type } = this.props
    const classes = classNames({
      [this.classesPrefix]: true,
      [`${className}`]: className,
    })
    const iconToRender = this.iconDefinition(type)

    return (
      <i className={classes} aria-label={`icon: ${type}`}>
        {React.createElement(iconToRender)}
      </i>
    )
  }
}

Icon.defaultProps = {
  type: 'conecto',
}

const IconTypePropTypes = PropTypes.oneOf([
  'alert',
  'arrow-down',
  'bookmark',
  'check',
  'close',
  'cog',
  'concurrents',
  'email',
  'facebook',
  'flag',
  'colored-flag',
  'marker',
  'menu',
  'notification',
  'phone',
  'search',
  'sms',
  'thumb-up',
  'user',
  'cc-amex',
  'cc-default',
  'cc-dinersclub',
  'cc-discover',
  'cc-jcb',
  'cc-mastercard',
  'cc-visa',
  'conecto',
])

Icon.propTypes = {
  type: IconTypePropTypes,
  className: PropTypes.string,
}

export { Icon as default, IconTypePropTypes }
