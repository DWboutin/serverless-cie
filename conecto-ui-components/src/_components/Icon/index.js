import Icon, { IconTypePropTypes } from './Icon'

export { Icon as default, IconTypePropTypes }
