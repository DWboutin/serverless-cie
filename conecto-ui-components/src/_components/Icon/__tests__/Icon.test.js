import React from 'react'
import { shallow, mount } from 'enzyme'

import Icon from '../../Icon'
import Facebook from '../../Icon/Icons/Facebook'
import Menu from '../../Icon/Icons/Menu'

global.console = {
  error: jest.fn(),
}

describe('Icon', () => {
  describe('SVG components', () => {
    it('Facebook SVG component should match snapshot', () => {
      const wrapper = shallow(<Facebook />)

      expect(wrapper).toMatchSnapshot()
    })

    it('Menu icon component should match snapshot', () => {
      const wrapper = shallow(<Menu />)

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('Default', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should contain the added classes', () => {
      const wrapper = shallow(<Icon className="test-class-1 test-class-2" />)

      expect(wrapper.hasClass('test-class-1')).toEqual(true)
      expect(wrapper.hasClass('test-class-2')).toEqual(true)
    })

    it('should render the short logo is default icon', () => {
      const wrapper = mount(<Icon />)

      expect(wrapper.find('svg').length).toEqual(1)
      expect(wrapper.find('svg').hasClass('to-logo-to')).toEqual(true)

      wrapper.unmount()
    })
  })

  describe('type="alert"', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon type="alert" />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon type="alert" />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render the Alert icon', () => {
      const wrapper = mount(<Icon type="alert" />)

      expect(wrapper.find('svg').length).toEqual(1)
      expect(wrapper.find('svg').hasClass('to-icon-alert')).toEqual(true)

      wrapper.unmount()
    })
  })

  describe('type="arrow-down"', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon type="arrow-down" />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon type="arrow-down" />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render the Arrow Down icon', () => {
      const wrapper = mount(<Icon type="arrow-down" />)

      expect(wrapper.find('svg').length).toEqual(1)
      expect(wrapper.find('svg').hasClass('to-icon-arrow-down')).toEqual(true)

      wrapper.unmount()
    })
  })

  describe('type="bookmark"', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon type="bookmark" />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon type="bookmark" />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render the Bookmark icon', () => {
      const wrapper = mount(<Icon type="bookmark" />)

      expect(wrapper.find('svg').length).toEqual(1)
      expect(wrapper.find('svg').hasClass('to-icon-bookmark')).toEqual(true)

      wrapper.unmount()
    })
  })

  describe('type="check"', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon type="check" />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon type="check" />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render the Check icon', () => {
      const wrapper = mount(<Icon type="check" />)

      expect(wrapper.find('svg').length).toEqual(1)
      expect(wrapper.find('svg').hasClass('to-icon-check')).toEqual(true)

      wrapper.unmount()
    })
  })

  describe('type="close"', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon type="close" />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon type="close" />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render the Close icon', () => {
      const wrapper = mount(<Icon type="close" />)

      expect(wrapper.find('svg').length).toEqual(1)
      expect(wrapper.find('svg').hasClass('to-icon-close')).toEqual(true)

      wrapper.unmount()
    })
  })

  describe('type="cog"', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon type="cog" />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon type="cog" />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render the Cog icon', () => {
      const wrapper = mount(<Icon type="cog" />)

      expect(wrapper.find('svg').length).toEqual(1)
      expect(wrapper.find('svg').hasClass('to-icon-cog')).toEqual(true)

      wrapper.unmount()
    })
  })

  describe('type="concurrents"', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon type="concurrents" />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon type="concurrents" />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render the Concurrents icon', () => {
      const wrapper = mount(<Icon type="concurrents" />)

      expect(wrapper.find('svg').length).toEqual(1)
      expect(wrapper.find('svg').hasClass('to-icon-concurrents')).toEqual(true)

      wrapper.unmount()
    })
  })

  describe('type="email"', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon type="email" />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon type="email" />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render the Email icon', () => {
      const wrapper = mount(<Icon type="email" />)

      expect(wrapper.find('svg').length).toEqual(1)
      expect(wrapper.find('svg').hasClass('to-icon-email')).toEqual(true)

      wrapper.unmount()
    })
  })

  describe('type="facebook"', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon type="facebook" />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon type="facebook" />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render the Facebook icon', () => {
      const wrapper = mount(<Icon type="facebook" />)

      expect(wrapper.find('svg').length).toEqual(1)
      expect(wrapper.find('svg').hasClass('to-icon-facebook')).toEqual(true)

      wrapper.unmount()
    })
  })

  describe('type="flag"', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon type="flag" />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon type="flag" />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render the Flag icon', () => {
      const wrapper = mount(<Icon type="flag" />)

      expect(wrapper.find('svg').length).toEqual(1)
      expect(wrapper.find('svg').hasClass('to-icon-flag')).toEqual(true)

      wrapper.unmount()
    })
  })

  describe('type="colored-flag"', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon type="colored-flag" />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon type="colored-flag" />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render the Colored Flag icon', () => {
      const wrapper = mount(<Icon type="colored-flag" />)

      expect(wrapper.find('svg').length).toEqual(1)
      expect(wrapper.find('svg').hasClass('to-icon-colored-flag')).toEqual(true)

      wrapper.unmount()
    })
  })

  describe('type="marker"', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon type="marker" />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon type="marker" />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render the Flag icon', () => {
      const wrapper = mount(<Icon type="marker" />)

      expect(wrapper.find('svg').length).toEqual(1)
      expect(wrapper.find('svg').hasClass('to-icon-marker')).toEqual(true)

      wrapper.unmount()
    })
  })

  describe('type="menu"', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon type="menu" />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon type="menu" />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render the Menu icon', () => {
      const wrapper = mount(<Icon type="menu" />)

      expect(wrapper.find('div').length).toEqual(1)
      expect(wrapper.find('div').hasClass('to-icon-menu')).toEqual(true)
      expect(wrapper.find('span').length).toEqual(2)

      wrapper.unmount()
    })
  })

  describe('type="notification"', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon type="notification" />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon type="notification" />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render the Notification icon', () => {
      const wrapper = mount(<Icon type="notification" />)

      expect(wrapper.find('svg').length).toEqual(1)
      expect(wrapper.find('svg').hasClass('to-icon-notification')).toEqual(true)

      wrapper.unmount()
    })
  })

  describe('type="phone"', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon type="phone" />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon type="phone" />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render the Phone icon', () => {
      const wrapper = mount(<Icon type="phone" />)

      expect(wrapper.find('svg').length).toEqual(1)
      expect(wrapper.find('svg').hasClass('to-icon-phone')).toEqual(true)

      wrapper.unmount()
    })
  })

  describe('type="search"', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon type="search" />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon type="search" />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render the Search icon', () => {
      const wrapper = mount(<Icon type="search" />)

      expect(wrapper.find('svg').length).toEqual(1)
      expect(wrapper.find('svg').hasClass('to-icon-search')).toEqual(true)

      wrapper.unmount()
    })
  })

  describe('type="sms"', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon type="sms" />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon type="sms" />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render the SMS icon', () => {
      const wrapper = mount(<Icon type="sms" />)

      expect(wrapper.find('svg').length).toEqual(1)
      expect(wrapper.find('svg').hasClass('to-icon-sms')).toEqual(true)

      wrapper.unmount()
    })
  })

  describe('type="thumb-up"', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon type="thumb-up" />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon type="thumb-up" />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render the Thumb Up icon', () => {
      const wrapper = mount(<Icon type="thumb-up" />)

      expect(wrapper.find('svg').length).toEqual(1)
      expect(wrapper.find('svg').hasClass('to-icon-thumb-up')).toEqual(true)

      wrapper.unmount()
    })
  })

  describe('type="user"', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon type="user" />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon type="user" />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render the User icon', () => {
      const wrapper = mount(<Icon type="user" />)

      expect(wrapper.find('svg').length).toEqual(1)
      expect(wrapper.find('svg').hasClass('to-icon-user')).toEqual(true)

      wrapper.unmount()
    })
  })

  describe('type="cc-amex"', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon type="cc-amex" />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon type="cc-amex" />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render the CC-Amex icon', () => {
      const wrapper = mount(<Icon type="cc-amex" />)

      expect(wrapper.find('svg').length).toEqual(1)
      expect(wrapper.find('svg').hasClass('to-icon-cc-amex')).toEqual(true)

      wrapper.unmount()
    })
  })

  describe('type="cc-default"', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon type="cc-default" />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon type="cc-default" />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render the CC-Default icon', () => {
      const wrapper = mount(<Icon type="cc-default" />)

      expect(wrapper.find('svg').length).toEqual(1)
      expect(wrapper.find('svg').hasClass('to-icon-cc-default')).toEqual(true)

      wrapper.unmount()
    })
  })

  describe('type="cc-dinersclub"', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon type="cc-dinersclub" />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon type="cc-dinersclub" />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render the CC-DinersClub icon', () => {
      const wrapper = mount(<Icon type="cc-dinersclub" />)

      expect(wrapper.find('svg').length).toEqual(1)
      expect(wrapper.find('svg').hasClass('to-icon-cc-dinersclub')).toEqual(
        true
      )

      wrapper.unmount()
    })
  })

  describe('type="cc-discover"', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon type="cc-discover" />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon type="cc-discover" />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render the CC-Discover icon', () => {
      const wrapper = mount(<Icon type="cc-discover" />)

      expect(wrapper.find('svg').length).toEqual(1)
      expect(wrapper.find('svg').hasClass('to-icon-cc-discover')).toEqual(true)

      wrapper.unmount()
    })
  })

  describe('type="cc-jcb"', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon type="cc-jcb" />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon type="cc-jcb" />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render the CC-JCB icon', () => {
      const wrapper = mount(<Icon type="cc-jcb" />)

      expect(wrapper.find('svg').length).toEqual(1)
      expect(wrapper.find('svg').hasClass('to-icon-cc-jcb')).toEqual(true)

      wrapper.unmount()
    })
  })

  describe('type="cc-mastercard"', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon type="cc-mastercard" />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon type="cc-mastercard" />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render the CC-Mastercard icon', () => {
      const wrapper = mount(<Icon type="cc-mastercard" />)

      expect(wrapper.find('svg').length).toEqual(1)
      expect(wrapper.find('svg').hasClass('to-icon-cc-mastercard')).toEqual(
        true
      )

      wrapper.unmount()
    })
  })

  describe('type="cc-visa"', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon type="cc-visa" />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon type="cc-visa" />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render the CC-Visa icon', () => {
      const wrapper = mount(<Icon type="cc-visa" />)

      expect(wrapper.find('svg').length).toEqual(1)
      expect(wrapper.find('svg').hasClass('to-icon-cc-visa')).toEqual(true)

      wrapper.unmount()
    })
  })

  describe('type="conecto"', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon type="conecto" />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon type="conecto" />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render the Conecto icon', () => {
      const wrapper = mount(<Icon type="conecto" />)

      expect(wrapper.find('svg').length).toEqual(1)
      expect(wrapper.find('svg').hasClass('to-logo-to')).toEqual(true)

      wrapper.unmount()
    })
  })

  describe('no type', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render the Conecto icon by default', () => {
      const wrapper = mount(<Icon />)

      expect(wrapper.find('svg').length).toEqual(1)
      expect(wrapper.find('svg').hasClass('to-logo-to')).toEqual(true)

      wrapper.unmount()
    })
  })

  it('should console.error when passing a number to type prop', () => {
    shallow(<Icon type={123}>Test</Icon>)

    expect(global.console.error).toBeCalled()
  })

  it('should console.error when not passing a good value to type prop', () => {
    shallow(<Icon type="test">Test</Icon>)

    expect(global.console.error).toBeCalled()
  })
})
