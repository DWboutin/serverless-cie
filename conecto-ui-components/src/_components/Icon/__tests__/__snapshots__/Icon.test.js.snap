// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Default should match snapshot 1`] = `
<i
  aria-label="icon: conecto"
  className="to-icon"
>
  <To />
</i>
`;

exports[`Icon SVG components Facebook SVG component should match snapshot 1`] = `
<svg
  className="to-icon-facebook"
  height="13"
  viewBox="0 0 13 13"
  width="13"
>
  <path
    d="M12.9078 0.737589V12.2624C12.9078 12.6312 12.6312 13 12.1702 13H8.94326V7.92908H10.6028L10.8794 5.99291H8.94326V4.70213C8.94326 4.14894 9.12766 3.78014 9.95745 3.78014H10.9716V2.02837C10.7872 2.02837 10.1418 1.93617 9.49645 1.93617C8.02128 1.93617 7.00709 2.85816 7.00709 4.51773V5.99291H5.25532V7.92908H6.91489V12.9078H0.737589C0.276596 12.9078 0 12.539 0 12.1702V0.737589C0 0.276596 0.276596 0 0.737589 0H12.2624C12.6312 0 12.9078 0.276596 12.9078 0.737589Z"
    fill="#572D2D"
  />
</svg>
`;

exports[`Icon SVG components Menu icon component should match snapshot 1`] = `
<div
  className="to-icon-menu"
>
  <span />
  <span />
</div>
`;

exports[`Icon no type should match snapshot 1`] = `
<i
  aria-label="icon: conecto"
  className="to-icon"
>
  <To />
</i>
`;

exports[`Icon type="alert" should match snapshot 1`] = `
<i
  aria-label="icon: alert"
  className="to-icon"
>
  <Alert />
</i>
`;

exports[`Icon type="arrow-down" should match snapshot 1`] = `
<i
  aria-label="icon: arrow-down"
  className="to-icon"
>
  <ArrowDown />
</i>
`;

exports[`Icon type="bookmark" should match snapshot 1`] = `
<i
  aria-label="icon: bookmark"
  className="to-icon"
>
  <Bookmark />
</i>
`;

exports[`Icon type="cc-amex" should match snapshot 1`] = `
<i
  aria-label="icon: cc-amex"
  className="to-icon"
>
  <CCAmex />
</i>
`;

exports[`Icon type="cc-default" should match snapshot 1`] = `
<i
  aria-label="icon: cc-default"
  className="to-icon"
>
  <CCDefault />
</i>
`;

exports[`Icon type="cc-dinersclub" should match snapshot 1`] = `
<i
  aria-label="icon: cc-dinersclub"
  className="to-icon"
>
  <CCDinersClub />
</i>
`;

exports[`Icon type="cc-discover" should match snapshot 1`] = `
<i
  aria-label="icon: cc-discover"
  className="to-icon"
>
  <CCDiscover />
</i>
`;

exports[`Icon type="cc-jcb" should match snapshot 1`] = `
<i
  aria-label="icon: cc-jcb"
  className="to-icon"
>
  <CCDiscover />
</i>
`;

exports[`Icon type="cc-mastercard" should match snapshot 1`] = `
<i
  aria-label="icon: cc-mastercard"
  className="to-icon"
>
  <CCMastercard />
</i>
`;

exports[`Icon type="cc-visa" should match snapshot 1`] = `
<i
  aria-label="icon: cc-visa"
  className="to-icon"
>
  <CCVisa />
</i>
`;

exports[`Icon type="check" should match snapshot 1`] = `
<i
  aria-label="icon: check"
  className="to-icon"
>
  <Check />
</i>
`;

exports[`Icon type="close" should match snapshot 1`] = `
<i
  aria-label="icon: close"
  className="to-icon"
>
  <Close />
</i>
`;

exports[`Icon type="cog" should match snapshot 1`] = `
<i
  aria-label="icon: cog"
  className="to-icon"
>
  <Cog />
</i>
`;

exports[`Icon type="colored-flag" should match snapshot 1`] = `
<i
  aria-label="icon: colored-flag"
  className="to-icon"
>
  <ColoredFlag />
</i>
`;

exports[`Icon type="concurrents" should match snapshot 1`] = `
<i
  aria-label="icon: concurrents"
  className="to-icon"
>
  <Concurrents />
</i>
`;

exports[`Icon type="conecto" should match snapshot 1`] = `
<i
  aria-label="icon: conecto"
  className="to-icon"
>
  <To />
</i>
`;

exports[`Icon type="email" should match snapshot 1`] = `
<i
  aria-label="icon: email"
  className="to-icon"
>
  <Email />
</i>
`;

exports[`Icon type="facebook" should match snapshot 1`] = `
<i
  aria-label="icon: facebook"
  className="to-icon"
>
  <Facebook />
</i>
`;

exports[`Icon type="flag" should match snapshot 1`] = `
<i
  aria-label="icon: flag"
  className="to-icon"
>
  <Flag />
</i>
`;

exports[`Icon type="marker" should match snapshot 1`] = `
<i
  aria-label="icon: marker"
  className="to-icon"
>
  <Marker />
</i>
`;

exports[`Icon type="menu" should match snapshot 1`] = `
<i
  aria-label="icon: menu"
  className="to-icon"
>
  <Menu />
</i>
`;

exports[`Icon type="notification" should match snapshot 1`] = `
<i
  aria-label="icon: notification"
  className="to-icon"
>
  <Notification />
</i>
`;

exports[`Icon type="phone" should match snapshot 1`] = `
<i
  aria-label="icon: phone"
  className="to-icon"
>
  <Phone />
</i>
`;

exports[`Icon type="search" should match snapshot 1`] = `
<i
  aria-label="icon: search"
  className="to-icon"
>
  <Search />
</i>
`;

exports[`Icon type="sms" should match snapshot 1`] = `
<i
  aria-label="icon: sms"
  className="to-icon"
>
  <SMS />
</i>
`;

exports[`Icon type="thumb-up" should match snapshot 1`] = `
<i
  aria-label="icon: thumb-up"
  className="to-icon"
>
  <ThumbUp />
</i>
`;

exports[`Icon type="user" should match snapshot 1`] = `
<i
  aria-label="icon: user"
  className="to-icon"
>
  <User />
</i>
`;
